[libc] Add bind function

michaelrj-google opened the pull request:
This patch adds the bind function to go with the socket function. It
also cleans up a lot of socket related data structures.

michaelrj-google committed:
[libc] Add bind function

This patch adds the bind function to go with the socket function. It
also cleans up a lot of socket related data structures.

lntue reviewed (commented):
It looks like real error from the buildkite log.

https://buildkite.com/llvm-project/github-pull-requests/builds/19127#018c22db-3c2f-416d-a07b-6e508784b9af/6-6313

michaelrj-google committed:
move socket functions to fullbuild

They should have been in fullbuild only from the start, I'm not sure how
I missed that they weren't.

michaelrj-google commented:
the buildkite error should be fixed now
